i6_models/assemblies/lstm/lstm_v1.py

```diff
@@ -0,0 +1,70 @@
+__all__ = [
+    "LstmEncoderV1Config",
+    "LstmEncoderV1",
+]
+
+from dataclasses import dataclass
+import torch
+from torch import nn
+from typing import Any, Dict, Optional, Tuple, Union
+
+from i6_models.config import ModelConfiguration
+from i6_models.parts.lstm import LstmBlockV1Config, LstmBlockV1
+
+
+@dataclass
+class LstmEncoderV1Config(ModelConfiguration):
+    input_dim: int
+    embed_dim: int
+    embed_dropout: float
+    lstm_layers_cfg: LstmBlockV1Config
+    lstm_dropout: float
+    init_args: Optional[Dict[str, Any]] = None
+
+    @classmethod
+    def from_dict(cls, model_cfg_dict: Dict):
```

Comment:
I haven't seen these functions for other config classes. What is the reason they're needed here? Should we add generic "serialization" support to all other parameter dataclasses as well?

Reply:
I think this depends on how you import/call your model and how the params are then passed to the model.. 
I like to have a dataclass in order to set the params in my config. and I like to have a single attribute `cfg` instead of numerous attributes for the model..
But passing a dict instead of the dataclass allows me to write someting like this in the returnn config without the need to import the data classes:
```
get_model = __import__("functools").partial(
    __import__(
        "apptek_asr.users.cluescher.lib.returnn.pytorch_networks.lm.lstm.alex_variant_v6",
        fromlist=["Model"],
    ).Model,
    **{
        "model_config_dict": {
            "vocab_dim": 249352,
            "embed_dim": 128,
            "hidden_dim": 512,
            "n_lstm_layers": 2,
            "init_args": {
                "init_args_w": {"func": "normal", "arg": {"mean": 0.0, "std": 0.1}},
                "init_args_b": {"func": "normal", "arg": {"mean": 0.0, "std": 0.1}},
            },
            "num_noise": 25,
            "bias": True,
            "bottle_neck_dim": 512,
            "dropout": 0.2,
        }
    }
)
```
maybe needed is therefor a bit strong.. :D